### PR TITLE
chore(release): add shipjs

### PR DESCRIPTION
**Pull request by eunjae-lee**
**Summary**

This PR adds shipjs as a release tool.

**Result**

Running `yarn run release` creates a pull request for release.

### CODE REVIEW

File: ship.config.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      version,
+      tagName,
+      latestCommitHash,
+      latestCommitUrl,
+      repoURL,
+    }) => ({
+      pretext: [
+        `:tada: Successfully released *${appName}@${version}*`,
+        '',
+        `Make sure to run \`yarn run update_examples ${version}\` in \`react-instantsearch\`.`,
```
Comment (Haroenv) [inline ship.config.js:64]: can we wait until the packages have been published and run the script here?

File: package.json [hunk 2]
```diff
@@ -29,7 +29,8 @@
     "test:size": "bundlesize",
     "storybook": "start-storybook -p 6006 -c storybook -s storybook/public",
     "storybook:build": "build-storybook -s storybook/public -o website/storybook -c storybook",
-    "website:build": "yarn build && yarn webpack --config website/webpack.config.js"
+    "website:build": "yarn build && yarn webpack --config website/webpack.config.js",
+    "update_examples": "node scripts/update-examples.js"
```
Comment (Haroenv) [inline package.json:33]: this one should maybe be release:examples and ordered after `release`

File: ship.config.js (hunk 1 above)
Comment (eunjae-lee) [reply ship.config.js]: https://github.com/algolia/react-instantsearch/pull/3017/commits/77ce2543ce8a9b87c3b4f6f31b67c0d56bf992b9 
WDTY?

**Review by Haroenv — APPROVED** [review]

**Comment by eunjae-lee** [comment]
Circle CI has been configured, btw.

**Pull request merged**